pyaedt/edb.py

```diff
@@ -1356,36 +1356,122 @@ def solve_siwave(self):
         return True
 
     @pyaedt_function_handler()
-    def add_design_variable(self, variable_name, variable_value):
+    def add_design_variable(self, variable_name, variable_value, is_parameter=False):
         """Add a design variable.
 
         Parameters
         ----------
         variable_name : str
-            Name of the variable.
+            Name of the variable, if it begins by a "$", it will be a "Project Variable".
         variable_value : str, float
             Value of the variable with units.
+        is_parameter : bool, optional
+            True to add a "Parameter Default", False for a "Local Variable"
 
         Returns
         -------
         tuple
             tuple containing the ``AddVariable`` result and variable server.
+
+        Examples
+        --------
+
+        >>> from pyaedt import Edb
+        >>> edb_app = Edb()
+        >>> boolean_1, ant_length = edb_app.add_design_variable("my_local_variable", "1cm")
+        >>> boolean_2, para_length = edb_app.change_design_variable_value("my_parameter", "1m", is_parameter=True
+        >>> boolean_3, project_length = edb_app.change_design_variable_value("$my_project_variable", "1m")
+
+
         """
-        is_parameter = True
         if "$" in variable_name:
-            var_server = self.db.GetVariableServer()
-            is_parameter = False
+            if variable_name.index("$") == 0:
+                var_server = self.db.GetVariableServer()
+                is_parameter = False
+                string_message = ["Creating Project Variable %s", "Project Variable %s exists. Using it."]
+            else:
+                var_server = self.active_cell.GetVariableServer()
+                self.logger.warning(
+                    "The character $ must be placed at the beginning of your variable name,"
+                    " to make it a Project Variable"
+                )
+
+                string_message = ["Creating Local Variable %s", "Local Variable %s exists. Using it."]
```

Comment:
Same type of editorial changes should be made here as in line 1392.

Reply:
I can add a point after the %s but not replace the "," by a "." string_message is a list what it contains must be separated by a comma.

Reply:
@jeremich41 Thanks for explaining.